bee-hive/bee_hive/workflow.py

```diff
@@ -39,15 +39,19 @@ def run(self):
             print("not supported yet")   
 
     def _sequence(self):
-        prompt = self.workflow["spec"]["template"]["prompt"]
-        for agent in self.agents.values():
-            if (
-                self.workflow["spec"]["strategy"]["output"]
-                and self.workflow["spec"]["strategy"]["output"] == "verbose"
-            ):
-                prompt = agent.run_streaming(prompt)
-            else:
-                prompt = agent.run(prompt)
+        prompt = self.workflow["spec"]["prompt"]
+        steps = self.workflow["spec"]["steps"]
+        for step in steps:
+            step_name = step["name"]
+            agent_name = step["agent"]
+            agent_instance = self.agents.get(agent_name)
+            if not agent_instance:
+                raise ValueError(f"Agent {agent_name} not found for step {step_name}")
+            self.steps[step_name] = Step({"name": step_name, "agent": agent_instance})
+
+        for step_name, step_obj in self.steps.items():
+            response = step_obj.run(prompt)  
+            prompt = response.get("prompt", prompt)
```

Comment:
prompt is currently not required by spec - so either this should be optional, or spec should be update to state as required

Reply:
Ok if we want to defer, as really we need full spec validation, or much more defensive code for parsing the objects.